File: insertion.c
/*
Name : Khan Arshad Abdulla
Roll No : 20CO24
*/

#include<stdio.h>
void main(){
    int n,i,j;
    printf("Enter size of array: ");
    scanf("%d",&n);
    int ar[n];
    printf("Enter Elements of Array: ");
    for(i=0;i<n;i++){
        scanf("%d",&ar[i]);
    }
    for(i=1;i<n;i++){
        int cur=ar[i];
        j=i-1;
        while(j>=0 && ar[j]>cur){
            ar[j+1]=ar[j];
            j--;
        }
        ar[j+1]=cur;
    }
    printf("Sorted  Array:: ");
    for(i=0;i<n;i++){
        printf("%d\t",ar[i]);
    }
}
/*
Output:

Enter size of array: 5
Enter Elements of Array: 9
2
8
5
7
Sorted  Array:: 2       5       7       8       9
*/

File: insertion.py
def insertion(list):
    for i in range(1,len(list)):
        curr = list[i]
        j=i-1
        while j>=0 and list[j]>curr:
            list[j+1]=list[j]
            j=j-1
        list[j+1]=curr    
            
    
def main():
    list = []
    n= int(input("Enter No of Elements: "))
    print("Enter Lists Elements: ")
    for i in range(0,n):
        a = int(input())
        list.append(a)
    insertion(list)
    print(list)
        
        
if __name__=='__main__':
    main()